docs/spec/manifest-v2-2.md

```diff
@@ -0,0 +1,255 @@
+# Image Manifest Version 2, Schema 2
+
+This document outlines the format of of the V2 image manifest, schema version 2.
+The original (and provisional) image manifest for V2 (schema 1), was introduced
+in the Docker daemon in the [v1.3.0
+release](https://github.com/docker/docker/commit/9f482a66ab37ec396ac61ed0c00d59122ac07453)
+and is specified in the [schema 1 manifest definition](./manifest-v2-1.md)
+
+This second schema version has two primary goals. The first is to allow
+multi-architecture images, through a "fat manifest" which references image
```

Comment:
`docker build` will work as it does today creating a single architecture image.  `docker push` is what creates manifests today, and the discussion will have to happen around whether we can "tuck" the features of creating a multi-arch manifest (and reference/push layers for more than one architecture image) in the flow of `docker push` or will `docker push` create a traditional single-architecture manifest, and that digest hash will be used in a command like `docker manifest [create]`, along with other digest hashes to construct the 'fat manifest' after all single-architecture manifests/layers are pushed?

We had a discussion about this briefly with @RichardScothern and @stevvooe here at DockerCon, and I hope to, with @harche, create some user flows and add them as a comment in #200 (or wherever makes sense) as it is broader than just the manifest schema format itself--which is this PR's focus.  Once we have those, that should give us a framework of actions/proposed commands/user flow to have a targeted discussion on what makes the most sense.